src/Transports/MassTransit.AmazonSqsTransport/Pipeline/AmazonSqsMessageReceiver.cs

```diff
@@ -170,5 +241,39 @@ async Task ActiveAndActualAgentsCompleted(StopContext context)
                 }
             }
         }
+
+        class SequenceNumberComparer : IComparer<string>
+        {
+            public static readonly SequenceNumberComparer Instance = new SequenceNumberComparer();
+
+            public int Compare(string x, string y)
+            {
+                if (string.IsNullOrEmpty(x))
+                {
+                    throw new ArgumentNullException(nameof(x));
+                }
+
+                if (string.IsNullOrEmpty(y))
+                {
+                    throw new ArgumentNullException(nameof(y));
+                }
+
+                if (x.Length != y.Length)

```

Comment:
This can be greatly simplified. If the length is greater, it's a bigger number. If the length is shorter, its a smaller number. If the length is equal, compare the string. Done. No need to parse it.

Reply:
length might be always the same, including zeros at the begging, at least kinesis is doing that. But not sure about SQS